File: composeApp/src/iosMain/kotlin/no/nordicsemi/nrf/matter/matter/MatterStorage.kt
package no.nordicsemi.nrf.matter.matter

import platform.Foundation.NSData
import platform.Foundation.NSUserDefaults
import platform.Matter.MTRStorageProtocol
import platform.darwin.NSObject

class MatterStorage : NSObject(), MTRStorageProtocol {
    private val defaults = NSUserDefaults(suiteName = "group.P3R8YQEV4L.nordicsemi.nrf.matter")

    override fun removeStorageDataForKey(key: String): Boolean {
        defaults.removeObjectForKey(key)
        return defaults.synchronize()
    }

    override fun storageDataForKey(key: String): NSData? {
        return defaults.dataForKey(key)
    }

    override fun setStorageData(value: NSData, forKey: String): Boolean {
        defaults.setObject(value, forKey)
        return defaults.synchronize()
    }
}

File: iosApp/SharedCode/MatterStorage.swift
//
//  MatterStorage.swift
//  iosApp
//
//  Created by Sylwester Zielinski on 05/03/2026.
//

import Matter
import os.log

public class MatterStorage : NSObject, MTRStorage {
    
    private let defaults = UserDefaults(suiteName: "group.nordicsemi.nrf.matter")!
    private let logger = Logger(subsystem: "nrf.matter", category: "MatterStorage")
    
    public func storeString(key: String, value: String) {
        defaults.set(key, forKey: key)
    }
    
    public func getString(key: String) -> String? {
        defaults.string(forKey: key)
    }
    
    public func storageData(forKey key: String) -> Data? {
        return defaults.data(forKey: key)
    }

    public func setStorageData(_ value: Data, forKey key: String) -> Bool {
        defaults.setValue(value, forKey: key)
        return true
    }

    public func removeStorageData(forKey key: String) -> Bool {
        defaults.removeObject(forKey: key)
        return true
    }
    
    public func getKey(forKey key: String) -> Data? {
        return defaults.data(forKey: key)
    }
    
    
    public func setKey(_ value: Data, forKey key: String) -> Bool {
        defaults.setValue(value, forKey: key)
        return true
    }
}
